### PR TITLE
[Feature] Textarea component

### PR DESCRIPTION
**Pull request by GuillaumeGuerin1**
### ❓ Types of changes
Add a textarea component

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality)
- [x] 📦 New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Zeroheight documentation: https://zeroheight.com/34f250d18/p/32dc76-text-area

### 📝 Checklist

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes
- [ ] The component exists on old Prestashop UIKit and my pull request on [migrating documentation](https://github.com/PrestaShopCorp/devdocs.uikit.prestashop.com) is accepted.

### CODE REVIEW

File: packages/components/textarea/src/textarea.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      class="puik-textarea__character-count"
+      :class="{
+        'puik-textarea__character-count--error': characterLength > maxlength,
+      }"
+    >
+      <span>{{ characterLength }}/{{ maxlength }}</span>
+    </div>
+    <div class="puik-textarea__wrapper" :class="textareaClass">
+      <textarea
+        :id="id"
```
Comment (Kelhel) [inline packages/components/textarea/src/textarea.vue:14]: Ta props id n'est pas obligatoire et est par défaut un texte vide, il ne faut pas d'id avec un string vide

File: packages/components/textarea/stories/textarea.stories.ts [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+import PuikTextarea from './../src/textarea.vue'
+import PuikLabel from './../../label/src/label.vue'
+import type { Meta, Story, Args } from '@storybook/vue3'
+
+export default {
+  title: 'Components/Textarea',
+  component: PuikTextarea,
+  argTypes: {
+    id: {
+      description: '',
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:11]: Il manque la description sur toutes les props

File: packages/components/textarea/stories/textarea.stories.ts [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+  },
+  setup() {
+    const value = ref('')
+    return { value, args }
+  },
+  template: `
+<puik-textarea
+  v-bind="args"
+  v-model="value">
+</puik-textarea>
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:136]:
```suggestion
<puik-textarea v-bind="args" v-model="value"></puik-textarea>
```

File: packages/components/textarea/stories/textarea.stories.ts [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+}
+
+export const Disabled = Template.bind({})
+Disabled.args = {
+  id: 'textarea-1',
+  name: '',
+  placeholder: '',
+  autofocus: false,
+  required: false,
+  success: false,
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:189]:
```suggestion
```

File: packages/components/textarea/stories/textarea.stories.ts [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+      code: `
+      <!--VueJS Snippet-->
+<puik-textarea
+  id="textarea"
+  disabled
+  v-model="value"
+>
+</puik-textarea>
+      
+      <!--HTML/CSS Snippet-->
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:210]:
```suggestion
<!--VueJS Snippet-->
<puik-textarea
  id="textarea"
  disabled
  v-model="value"
>
</puik-textarea>
      
<!--HTML/CSS Snippet-->
```

File: packages/components/textarea/stories/textarea.stories.ts [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+}
+
+export const Readonly = Template.bind({})
+Readonly.args = {
+  id: 'textarea-1',
+  name: '',
+  placeholder: '',
+  autofocus: false,
+  required: false,
+  success: false,
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:229]:
```suggestion
```

File: packages/components/textarea/stories/textarea.stories.ts [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+  autoGrow: true,
+  rows: 2,
+  maxRows: 2,
+  maxlength: undefined,
+}
+Readonly.parameters = {
+  docs: {
+    source: {
+      code: `
+      <!--VueJS Snippet-->
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:242]:
```suggestion
<!--VueJS Snippet-->
```

File: packages/components/textarea/stories/textarea.stories.ts [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+<puik-textarea
+  id="textarea"
+  readonly
+  v-model="value"
+>
+</puik-textarea>
+      
+      <!--HTML/CSS Snippet-->
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:250]:
```suggestion
<!--HTML/CSS Snippet-->
```

File: packages/components/textarea/stories/textarea.stories.ts [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+}
+
+export const Error = Template.bind({})
+Error.args = {
+  id: 'textarea-1',
+  name: '',
+  placeholder: '',
+  autofocus: false,
+  required: false,
+  success: true,
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:269]:
```suggestion
```

File: packages/components/textarea/stories/textarea.stories.ts [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+      code: `
+      <!--VueJS Snippet-->
+<puik-textarea
+  id="textarea"
+  error="This is an error message"
+  v-model="value"
+>
+</puik-textarea>
+      
+      <!--HTML/CSS Snippet-->
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:291]:
```suggestion
<!--HTML/CSS Snippet-->
```

File: packages/components/textarea/stories/textarea.stories.ts [hunk 11]
```diff
@@ -0,0 +1,400 @@
...
+}
+
+export const CharacterCount = Template.bind({})
+CharacterCount.args = {
+  id: 'textarea-1',
+  name: '',
+  placeholder: '',
+  autofocus: false,
+  required: false,
+  success: false,
```
Comment (Kelhel) [inline packages/components/textarea/stories/textarea.stories.ts:316]:
```suggestion
```

**Review by Kelhel — CHANGES_REQUESTED** [review]

File: packages/components/textarea/test/textarea.spec.ts [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+describe('Textarea tests', () => {
+  let wrapper: VueWrapper<any>
+  const factory = (
+    propsData: Record<string, any> = {},
+    options: MountingOptions<any> = {}
+  ) => {
+    wrapper = mount(PuikTextarea, {
+      props: {
+        ...propsData,
+        'onUpdate:modelValue': async (modelValue: any) =>
```
[the remark this answers is not in the document]
Comment (GuillaumeGuerin1) [reply packages/components/textarea/test/textarea.spec.ts:15]: Non, c'est bien `onUpdate`
https://test-utils.vuejs.org/guide/advanced/v-model.html#a-simple-example

**Review by cnavarro-prestashop — APPROVED** [review]

**Review by Kelhel — APPROVED** [review]

**Pull request merged**